Client/game_sa/CTxdPoolSA.cpp

```diff
@@ -0,0 +1,66 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *  LICENSE:     See LICENSE in the top level directory
+ *
+ *  Multi Theft Auto is available from https://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+
+#include "StdInc.h"
+#include "CTxdPoolSA.h"
+#include "CGameSA.h"
+#include "CKeyGenSA.h"
+
+extern CGameSA* pGame;
+
+CTxdPoolSA::CTxdPoolSA()
+{
+    m_ppTxdPoolInterface = (CPoolSAInterface<CTextureDictonarySAInterface>**)0xC8800C;
+}
+
+std::uint32_t CTxdPoolSA::AllocateTextureDictonarySlot(std::uint32_t uiSlotId, std::string& strTxdName)
+{
+    CTextureDictonarySAInterface* pTxd = (*m_ppTxdPoolInterface)->AllocateAt(uiSlotId);
+    if (!pTxd)
+        return -1;
+
+    strTxdName.resize(24);
+
+    pTxd->usUsagesCount = 0;
+    pTxd->hash = pGame->GetKeyGen()->GetUppercaseKey(strTxdName.c_str());
+    pTxd->rwTexDictonary = nullptr;
+    pTxd->usParentIndex = -1;
+
+    return (*m_ppTxdPoolInterface)->GetObjectIndex(pTxd);
+}
+
+void CTxdPoolSA::RemoveTextureDictonarySlot(std::uint32_t uiTxdId)
+{
+    if (!(*m_ppTxdPoolInterface)->IsContains(uiTxdId))
+        return;
+
+    typedef std::uint32_t(__cdecl * Function_TxdReleaseSlot)(std::uint32_t uiTxdId);
```

Comment:
`using` instead of `typedef`:
```cpp
using Function_TxdReleaseSlot = std::uint32_t(__cdecl*)(std::uint32_t uiTxdId);
```

Reply:
Refactor is unrelated to this PR. Im resolving all conversation related to CTxdPoolSA